lib/endpoints/class-wp-json-attachments-controller.php

```diff
@@ -0,0 +1,337 @@
+<?php
+
+class WP_JSON_Attachments_Controller extends WP_JSON_Posts_Controller {
+
+	/**
+	 * Create a single attachment
+	 *
+	 * @param WP_JSON_Request $request Full details about the request
+	 * @return WP_Error|WP_HTTP_ResponseInterface
+	 */
+	public function create_item( $request ) {
+
+		// Permissions check - Note: "upload_files" cap is returned for an attachment by $post_type_obj->cap->create_posts
+		$post_type_obj = get_post_type_object( $this->post_type );
+		if ( ! current_user_can( $post_type_obj->cap->create_posts ) || ! current_user_can( $post_type_obj->cap->edit_posts ) ) {
```

Comment:
I know we didn't merge #854 yet, but the `permissions_callback` already exists. Maybe we just do that on a second pass though?